clippy_lints/src/strings.rs

```diff
@@ -438,27 +438,88 @@ declare_clippy_lint! {
 
 declare_lint_pass!(StringToString => [STRING_TO_STRING]);
 
+fn is_parent_map_like(cx: &LateContext<'_>, expr: &Expr<'_>) -> Option<rustc_span::Span> {
+    if let Some(parent_expr) = get_parent_expr(cx, expr)
+        && let ExprKind::MethodCall(name, ..) = parent_expr.kind
+        && name.ident.name == sym::map
+        && let Some(caller_def_id) = cx.typeck_results().type_dependent_def_id(parent_expr.hir_id)
+        && (clippy_utils::is_diag_item_method(cx, caller_def_id, sym::Result)
+            || clippy_utils::is_diag_item_method(cx, caller_def_id, sym::Option)
+            || clippy_utils::is_diag_trait_item(cx, caller_def_id, sym::Iterator))
+    {
+        Some(parent_expr.span)
+    } else {
+        None
+    }
+}
+
+fn is_called_from_map_like(cx: &LateContext<'_>, expr: &Expr<'_>) -> Option<rustc_span::Span> {
+    let parent = get_parent_expr(cx, expr)?;
+
+    match parent.kind {
+        ExprKind::Closure(_) => is_parent_map_like(cx, parent),
+        ExprKind::Block(_, _) => is_called_from_map_like(cx, parent),
```

Comment:
What is it used for? Is that even possible to get a `Block` from `get_parent_expr()` (my reasoning: it goes through `parent_hir_node()` that would return a `Node::Block`, not a `Node::Expr`, but I may be wrong)?

In any case, removing this arm doesn't make any test fail, so it looks suspicious.

Reply:
I thought it was needed to handle `.map(|x| { x.to_string() })` but I can't reproduce it so dropping it.